packages/react-browser-pathname/src/context.ts

```diff
@@ -1,7 +1,14 @@
 import React, { useContext, useCallback, useEffect, useState } from 'react';
 import type { ReactNode } from 'react';
 
+export type SET_ACTION = {
+  url?: string;
+  title: string;
+  data: object;
```

Comment:
i'd avoid exposing data unless you plan to expose it in `onChange` and current location also